### PR TITLE
Added publish dialog

### PR DESCRIPTION
**Pull request by JordanTheriault**
Complete publish dialog, ready to be connected with backend functionality.

### CODE REVIEW

File: public/friendlycode/js/fc/bramble-ui-bridge.js [hunk 1]
```diff
@@ -184,6 +184,49 @@ define(["jquery"], function($) {
...
+    });
+
+    $("#publish-button-cancel").click(function() {
+      displayPublishDialog(false);
+    });
+    $("#publish-underlay").click(function() {
+      displayPublishDialog(false);
+    });
+
+    function displayPublishDialog(visible) {
```
Comment (humphd) [inline public/friendlycode/js/fc/bramble-ui-bridge.js]: Split this out into two functions: `showPublishDialog()` and `hidePublishDialog()`

File: public/friendlycode/js/fc/bramble-ui-bridge.js [hunk 2]
```diff
@@ -184,6 +184,49 @@ define(["jquery"], function($) {
...
+        $("#publish-dialog").fadeOut();
+      }
+    }
+
+    // Variable for the state of publishPublic toggle switch
+    var publishPublic = true;
+    $("#publish-public-gallery-toggle").click(function() {
+      publishPublic === false ? publishPublic = true : publishPublic = false;
+      setPublicGalleryToggle(publishPublic);
+
```
Comment (humphd) [inline public/friendlycode/js/fc/bramble-ui-bridge.js]: Remove this blank line

**Comment by gideonthomas** [comment]
let's use 2-space indents everywhere to make rebasing easier

File: public/friendlycode/js/fc/bramble-ui-bridge.js [hunk 3]
```diff
@@ -184,6 +184,49 @@ define(["jquery"], function($) {
...
+    });
+    function setPublicGalleryToggle(publishPublic) {
+      if(publishPublic === true) {
+
```
Comment (humphd) [inline public/friendlycode/js/fc/bramble-ui-bridge.js]: Remove

File: public/friendlycode/js/fc/bramble-ui-bridge.js [hunk 4]
```diff
@@ -184,6 +184,49 @@ define(["jquery"], function($) {
...
+        $("#publish-public-gallery-toggle>ellipse").animate({cx: 31}, 250);
+        $("#publish-public-gallery-toggle>ellipse").css("fill", "#06a050");
+      }
+      else if(publishPublic === false)
+      {
```
Comment (humphd) [inline public/friendlycode/js/fc/bramble-ui-bridge.js]: `} else {`

File: public/stylesheets/publish.less [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+	box-shadow: 0 3px 9px rgba(0,0,0,0.24);
+	z-index: 151;
+  	position: absolute;
+  	font-family: 'Open Sans', sans-serif;
+    font-size: 1em;
+    font-weight: lighter;
+    font-weight: 300;
+    width: 400px;
+    right: 20px;
+    top: 65px;
```
Comment (humphd) [inline public/stylesheets/publish.less]: Odd indents here

File: public/stylesheets/publish.less [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+	border: none;
+	-webkit-box-shadow: 0 3px 9px rgba(0,0,0,0.24);
+	-moz-box-shadow: 0 3px 9px rgba(0,0,0,0.24);
+	box-shadow: 0 3px 9px rgba(0,0,0,0.24);
+	z-index: 151;
+  	position: absolute;
+  	font-family: 'Open Sans', sans-serif;
+    font-size: 1em;
+    font-weight: lighter;
+    font-weight: 300;
```
Comment (humphd) [inline public/stylesheets/publish.less]: You don't need a unit here?

File: public/stylesheets/publish.less [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+    width: 400px;
+    right: 20px;
+    top: 65px;
+
+    a {
+    	text-decoration: none;
+    	font-weight: 800;
+    }
+
+  	#publish-title {
```
Comment (humphd) [inline public/stylesheets/publish.less]: Indents throughout this file need an audit.  4-spaces please.

File: public/stylesheets/publish.less [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+	color: black;
+	padding: 10px;
+}
+
+.publish-public-gallery {
+	margin-top: 10px;
+	margin-bottom: 10px;
+
+	div.publish-public-gallery-toggle {
+
```
Comment (humphd) [inline public/stylesheets/publish.less]: Why have this if blank?

File: public/stylesheets/publish.less [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+}
+
+.publish-button {
+	display: inline-block;
+	padding: 5px 20px 5px 20px;
+	cursor: pointer;
+	color: white;
+	-webkit-user-select: none; /* Chrome/Safari */
+	-moz-user-select: none; /* Firefox */
+	-ms-user-select: none; /* IE10+ */
```
Comment (humphd) [inline public/stylesheets/publish.less]: Unprefixed as last item.

File: views/publish.html [hunk 10]
```diff
@@ -0,0 +1,43 @@
+{% block publishdialog %}
+<div id="publish-underlay" class="hide"></div>
+<div id="publish-dialog" class="hide">
+	<div id="publish-title">
+		Publish your Project
+	</div>
+
+
```
Comment (humphd) [inline views/publish.html]: Remove extra whitespace

File: views/userbar.html [hunk 11]
```diff
@@ -75,10 +75,10 @@
           </div>
         </div>
         <div id="navbar-publish">
-          <a href="" class="navbar-publish-button">
+          <div href="" id="navbar-publish-button">
```
Comment (humphd) [inline views/userbar.html]: Remove `href=""`

File: public/stylesheets/publish.less (hunk 6 above)
Comment (JordanTheriault) [inline public/stylesheets/publish.less]: Not for weight, no.

File: public/friendlycode/js/fc/bramble-ui-bridge.js [hunk 12]
```diff
@@ -184,6 +184,49 @@ define(["jquery"], function($) {
...
+
+    $("#publish-button-cancel").click(function() {
+      displayPublishDialog(false);
+    });
+    $("#publish-underlay").click(function() {
+      displayPublishDialog(false);
+    });
+
+    function displayPublishDialog(visible) {
+      if(visible === true) {
```
Comment (gideonthomas) [inline public/friendlycode/js/fc/bramble-ui-bridge.js]: `if(visible) {...} else {...}`

**Comment by JordanTheriault** [comment]
r+ from @gideonthomas

**Pull request merged**